addons/sourcemod/scripting/json_test.sp

```diff
@@ -707,6 +726,7 @@ void it_should_concat_arrays()
     arr2.PushInt(2);
     arr2.PushBool(false);
     arr2.SetHidden(1, true);
+    arr2.PushObject(new JSON_Object());
```

Comment:
Since the objects are now not of equal length, this test would fail if I hadn't also fixed the hidden key index bug.